presto-spi/src/main/java/com/facebook/presto/spi/ConnectorIndexHandle.java

```diff
@@ -15,4 +15,5 @@
 
 public interface ConnectorIndexHandle
 {
+    String toString(ConnectorSession session);
```

Comment:
This breaks the abstraction to all of our handle classes.  They are designed to be opaque handles with no methods on them.  The handle implementation should be able to have a reasonable implementation of `toString` without the need for session.... specifically for this case where you want one of the implementations to render the internal constraint details, you should be able to render all of that without a session, since the connector must be aware of the data types ahead of time.  If you also, want to render into the end user's locale/TZ, you can capture that information into the handle when you produce it, but I don't think that is necessary.

Reply:
@dain Dain, it all comes from needing to print `TupleDomain` -> `Domain` -> `ValueSet` -> `EquatableValueSet` which needs to call `type.getObjectValue(session, entry.getBlock(), 0)`.

```
    @Override
    public String toString(ConnectorSession session)
    {
        return (whiteList ? "[ " : "EXCLUDES[ ") + entries.stream()
                .map(entry -> type.getObjectValue(session, entry.getBlock(), 0).toString())
                .collect(Collectors.joining(", ")) + " ]";
    }
```

I tried removing `session` parameter from `Type.getObjectValue`, but ran into an issue. TimestampType and TimeType use `session` when printing out legacy timestamps (which are the default).

Do you have an idea of how to solve this some other way?